Code/src/main/java/nl/utwente/viskell/ui/MainOverlay.java

```diff
@@ -111,4 +126,38 @@ public void setTouchVisible(boolean value) {
         this.touchOverlay.setVisible(value);
     }
 
+    public void showPreferences() {
+        if (this.preferences == null) {
+            this.preferences = new PreferencesWindow(this);
+            this.mainPane.setPreferences(this.preferences);
+        }
+
+        this.preferences.show();
+    }
+
+    public void showInspector() {
+        if (this.inspector == null) {
+            this.inspector = new InspectorWindow(this);
+        }
+
+        this.inspector.show();
+    }
+
+    /**
+     * Zooms the underlying main pane in/out with a ratio, up to reasonable limits. 
+     * @param ratio the additional zoom factor to apply.
+     */
+    private void zoom(double ratio) {
+        double scale = this.mainPane.getScaleX();
+
+        /* Limit zoom to reasonable range. */
+        if (scale <= 0.2 && ratio < 1) return;
+        if (scale >= 3 && ratio > 1) return;
+
+        this.mainPane.setScaleX(scale * ratio);
+        this.mainPane.setScaleY(scale * ratio);
+        this.mainPane.setTranslateX(this.mainPane.getTranslateX() * ratio);
+        this.mainPane.setTranslateY(this.mainPane.getTranslateY() * ratio);
```

Comment:
This seems a little feature envy-ish (only ever accesses and changes fields on `ToplevelPane`, doesn't even touch `MainOverlay`)